pkg/util/securitycontext/security_context.go

```diff
@@ -0,0 +1,23 @@
+package securitycontext
+
+import (
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/container"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/podtemplatespec"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/envvar"
+)
+
+const (
+	ManagedSecurityContextEnv = "MANAGED_SECURITY_CONTEXT"
+)
+
+func WithDefaultSecurityContextsModifications() (podtemplatespec.Modification, container.Modification) {
+	managedSecurityContext := envvar.ReadBool(ManagedSecurityContextEnv)
+	configureContainerSecurityContext := container.NOOP()
+	configurePodSpecSecurityContext := podtemplatespec.NOOP()
+	if !managedSecurityContext {
+		configurePodSpecSecurityContext = podtemplatespec.WithSecurityContext(podtemplatespec.DefaultPodSecurityContext())
+		configureContainerSecurityContext = container.WithSecurityContext(container.DefaultSecurityContext())
+	}
+
+	return configurePodSpecSecurityContext, configureContainerSecurityContext
+}
```

Comment:
Q: Can this function be put in `podspec_template.go` instead of a package of its own?

Reply:
Sure, that shouldn't be problem.